File: app/src/main/java/pers/cxd/commonmodule/activities/splash/SplashPresenter.java
package pers.cxd.commonmodule.activities.splash;

import io.reactivex.rxjava3.annotations.NonNull;
import pers.cxd.rxlibrary.BaseObserverImpl;
import pers.cxd.rxlibrary.RxUtil;

class SplashPresenter extends SplashContract.Presenter{

    @Override
    void getSomeData(String arg1, String arg2) {
        if (mModel.someDataModelAvailable()){
            RxUtil.execute(mModel.someDataModel(arg1, arg2), new BaseObserverImpl<Object>(mSubscription) {
                @Override
                public void onNext(@io.reactivex.rxjava3.annotations.NonNull Object o) {
                    if (notDetach()){
                        mView.onSomeDataGotten(o);
                    }
                }

                @Override
                public void onError(@NonNull Throwable e) {

                }

                @Override
                public void onComplete() {
                    super.onComplete();
                    if (notDetach()){
                        mModel.clearSomeDataModel();
                    }
                }
            }, RxUtil.Transformers.IOToMain());
        }
    }

    @Override
    void register(String accountName, String password) {
        RxUtil.execute(mModel.registerModel(accountName, password), new BaseObserverImpl<Void>(mSubscription) {
            @Override
            public void onNext(@NonNull Void aVoid) {

            }

            @Override
            public void onError(@NonNull Throwable e) {

            }
        }, RxUtil.Transformers.IOToMain());
    }
}

File: app/src/main/java/pers/cxd/commonmodule/activities/splash/SplashPresenter.kt
package pers.cxd.commonmodule.activities.splash

import android.util.Log
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

internal object SplashPresenter : SplashContract.Presenter() {

    private const val TAG = "CXD-SplashPresenter"

    override fun test() {
        lifecycleScope.launch(Dispatchers.Main) {
            val result = mModel!!.testModel()
            Log.i(TAG, "test: $result")
            if (notDetach()) {
                if (result.isSuccess) {
                    mView!!.updateTestUI(result.getOrNull()!!)
                }
            }
        }
    }

}